service/builder/exporters_builder.go

```diff
@@ -29,22 +29,23 @@ import (
 // builtExporter is an exporter that is built based on a config. It can have
 // a trace and/or a metrics consumer and have a shutdown function.
 type builtExporter struct {
-	te component.TraceExporterBase
-	me component.MetricsExporterBase
+	logger *zap.Logger
```

Comment:
Do we need the logger to be here? It seems passing it as a parameter worked as well. I may be missing something but I do not yet see what we gain by doing this. It does not appear to be used in many places.
Unless it is a logger with custom fields, but I don't see that in the code.

Reply:
See the next comment about what logger we put here. We attach kind/type/name to the logger so all the entries will have that.